plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -132,7 +132,11 @@ public DataStore initialize(Map<String, Object> dsInfos) {
         Long zoneId = (Long)dsInfos.get("zoneId");
         String url = (String)dsInfos.get("url");
         String providerName = (String)dsInfos.get("providerName");
-        String hypervisorType = (String)dsInfos.get("hypervisorType");
+        Object hypervisorTypeObj = dsInfos.get("hypervisorType");
```

Comment:
it would be better to define hypervisorType as
```
HypervisorType hypervisorType = (String)dsInfos.get("hypervisorType");
```
the other lines in this file needs to be changed accordingly.

Reply:
I will update the PR.